app/assets/javascripts/provider/admin/apiconfig/services/integration.js.coffee

```diff
@@ -208,17 +208,21 @@ $(document).on 'initialize', '#proxy', ->
     false
 
   proxy_credentials_location_value = $('input[name="proxy[credentials_location]"]:checked').val()
+  credentials_forms = document.querySelectorAll('li[id^="proxy_auth_"]')
+  credentials_forms.forEach((el) -> el.classList.toggle('hidden', proxy_credentials_location_value is 'authorization'))
 
   replace_hyphens_or_underscores = (elt) ->
     return unless elt
     elt.value = switch proxy_credentials_location_value
       when 'headers' then elt.value.replace(/_/g,'-')
       when 'query' then elt.value.replace(/-/g,'_')
+      else elt.value
 
   $(document).on 'change', 'input[name="proxy[credentials_location]"]', (event) ->
     proxy_credentials_location_value = this.value
     replace_hyphens_or_underscores(document.getElementById('proxy_auth_user_key'))
     toggle_form_changed_ui()
+    credentials_forms.forEach((el) -> el.classList.toggle('hidden', proxy_credentials_location_value is 'authorization'))
```

Comment:
Is this line different from line 212 ?

Also. Why the forEach? You can just call toggleClass on the collection no?

Reply:
> Is this line different from line 212 ?
Same, will refactor.

> Also. Why the forEach? You can just call toggleClass on the collection no?
That's true for jQuery, not for browser native AFAIK. I thought the idea was to stop using jQuery gradually, do you think I should use it anyway?

Reply:
This whole file depends on jQuery. I don't see any benefit of not using it when you are doing so 3 lines above.

Reply:
I understand what you mean but it's the other way around: either we commit to make the codebase cleaner (even if it's little by little) or we keep adding legacy stuff. It's a matter of compromise and it will make it easier when we decide to refactor jQuery out for good.

Reply:
I don't agree that this one-liner is making the codebase cleaner. This file still depends on jQuery.

The code should be consistent and this file uses jQuery and has some style. Either change everything to follow some other style or match what is there already.

Reply:
IMO, either way is good about this precise piece of code, since we are going to ditch this file in the near future by a complete new implementation from scratch. I'd say let's not waste more time on this, the entire file and page is bloated and sucks... so..."what's another spot to a leopard?" what difference does it make? Well, none really. Let's move forward.  @josemigallas I'm happy either you use jquery or leave it as it is. I understand it will be more consistent keep coding the way it was, and also I reckon the work is already done.